src/reconcile/rec_write.c

```diff
@@ -2009,7 +2009,7 @@ __rec_compression_adjust(WT_SESSION_IMPL *session, uint32_t max, size_t compress
         else
             return;
     }
-    *adjustp = new;
+    WT_WRITE_ONCE(*adjustp, new);
```

Comment:
I'm not sure what this is protecting against. Are you concerned that the compiler could turn this into:

```
*adjustp = current;
*adjustp -= ten_percent;
```

Or something like that? That isn't harmful, since `adjustp` is set to `current` coming into the function.

Reply:
This is just aligning us with the comment above: `minor trickiness so we only read and write the value once.`

Reply:
To be clear we are being protected from what is described in your comment @agorrod